### PR TITLE
Feature/logic detail

### PR DESCRIPTION
**Pull request by at-phatnguyen**

### CODE REVIEW

File: MyApp/Model/Schema/CartItem.swift [hunk 1]
```diff
@@ -7,3 +7,36 @@
...
+
+final class CartItem: Object {
+    dynamic var id = 0
+    dynamic var name = ""
+    dynamic var price = 0
+    dynamic var imagePath = ""
+    dynamic var quantity: Int = 1
+
+    convenience init(menuItem: MenuItem, quantity: Int = 1) {
+        self.init()
```
Comment (at-tuanmai2) [inline MyApp/Model/Schema/CartItem.swift]: super.init không được à?

**Comment by at-daiho** [comment]
:+1: There are no linter violations.

File: MyApp/View/Controllers/Cart/CartViewController.swift [hunk 2]
```diff
@@ -38,39 +45,88 @@ class CartViewController: ViewController {
...
-        return rowHeight
-    }
-
     func tableView(_ tableView: UITableView, editActionsForRowAt indexPath: IndexPath) -> [UITableViewRowAction]? {
 
-        let deleteAction = UITableViewRowAction(style: .normal, title: App.Strings.delete, handler: { (_, _) in
-            print("Delete tapped")
+        let deleteAction = UITableViewRowAction(style: .normal, title: App.Strings.delete, handler: { [weak self] (_, _) in
+            guard let this = self else { return }
+            let refreshAlert = UIAlertController(title: "Xác nhận", message: "Xoá sản phẩm này?", preferredStyle: UIAlertControllerStyle.alert)
```
Comment (at-tuanmai2) [inline MyApp/View/Controllers/Cart/CartViewController.swift]: define

File: MyApp/View/Cell/CartCell/CartTableViewCell.swift [hunk 3]
```diff
@@ -8,7 +8,17 @@
 
 import UIKit
 
+protocol CartTableViewCellDelegate: class {
+    func cell(_ cell: CartTableViewCell, action performed: CartTableViewCell.Action)
```
Comment (at-tuanmai2) [inline MyApp/View/Cell/CartCell/CartTableViewCell.swift]: sai tên func rồi

File: MyApp/View/Controllers/Detail/DetailViewController.swift [hunk 4]
```diff
@@ -49,7 +57,7 @@ class DetailViewController: ViewController {
         for i in 0..<slideImages.count {
```
Comment (at-tuanmai2) [inline MyApp/View/Controllers/Detail/DetailViewController.swift]: user for slideImage in ...

File: MyApp/ViewModel/CartViewModel/CartViewModel.swift [hunk 5]
```diff
@@ -7,18 +7,72 @@
...
+        let currentItem = cartItems[indexPath.row]
+        let realm = RealmS()
+        realm.write {
+            realm.delete(currentItem)
+        }
+    }
+
+    func calculateSumMoney() -> Int {
+        var total = 0
+        for index in 0..<cartItems.count {
```
Comment (at-tuanmai2) [inline MyApp/ViewModel/CartViewModel/CartViewModel.swift]: use for cartItem in ....

File: MyApp/View/Cell/CartCell/CartTableViewCell.swift [hunk 6]
```diff
@@ -9,7 +9,7 @@
 import UIKit
 
 protocol CartTableViewCellDelegate: class {
-    func cell(_ cell: CartTableViewCell, action performed: CartTableViewCell.Action)
+    func actionForCell(_ cell: CartTableViewCell, action performed: CartTableViewCell.Action)
```
Comment (at-tuanmai2) [inline MyApp/View/Cell/CartCell/CartTableViewCell.swift]: func cell(_ cell: CartTableViewCell, needsPerformAction action: CartTableViewCell.Action)

**Pull request merged**